packages/grid/_modules_/grid/models/params/gridCellParams.ts

```diff
@@ -6,7 +6,7 @@ import { GridEditCellProps } from '../gridEditRowModel';
 /**
  * Object passed as parameter in the column [[GridColDef]] cell renderer.
  */
-export interface GridCellParams {
+export interface GridCellParams<T = GridCellValue, RowType = any> {
```

Comment:
We should limit the valid types


```suggestion
-export interface GridCellParams<T = GridCellValue, RowType = any> {
+export interface GridCellParams<T extends GridCellValue = GridCellValue, RowType = any> {
```

Reply:
Yes, nice. So what should be the generic type for handlers like `getCellClassName`?
```
getCellClassName?: (params: GridCellParams, details?: any) => string;
```
I am assuming `GridCellValue`. Like:
```
getCellClassName?: (params: GridCellParams<GridCellValue>, details?: any) => string;
```

`GridCellValue` does not support `any`.

Reply:
Ignore above comment, after limiting the valid types by applying above code suggestion when I add generic to params of `getCellClassName`, why am I getting this error?

![Screenshot (10)](https://user-images.githubusercontent.com/20900032/130782444-f18a4ff2-3109-40e9-ae6d-3d3002261f46.png)

Reply:
`getCellClassName` can be called with any cell of the grid

```tsx
<DataGrid
  rows={rows}
  columns={[{ field: 'id,  type: 'number }, { field: 'name', type: 'string }]
  getCellClassName={(params) => typeof params.value} // string or number depending on the column
 />
```

So I don't think users should override the param types like this

They could use a type guard function

```ts
const isNumberCell = (params: GridCellParams): params is GridCellParams<number> => params.field !== 'city'
```

Or cast with `as`

Reply:
I believe same happens for other methods, as cell value can be anything. And `GridCellParams` is used in callbacks only I think. So is extending with `GridCellValue` beneficial here? 
I think we need to support `any` then? as suggested here : https://github.com/mui-org/material-ui-x/issues/1329#issuecomment-814478259.

Reply:
Just saw this, if its relevant: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/78d27175abe1296b87089e44fedd7493abed9dca/types/react-table/index.d.ts#L311-L317

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/78d27175abe1296b87089e44fedd7493abed9dca/types/react-table/index.d.ts#L857